src/main/java/org/example/models/JobRole.java

```diff
@@ -27,19 +30,95 @@ public class JobRole {
     @JsonProperty
     private String status;
 
-    public JobRole(final int id, final String roleName,
-                   final String location,
-                   final String capability,
-                   final String band,
-                   final Date closingDate,
-                   final String status) {
-        this.id = id;
-        this.roleName = roleName;
-        this.location = location;
-        this.capability = capability;
-        this.band = band;
-        this.closingDate = closingDate;
-        this.status = status;
+    @JsonProperty
+    private String description;
+
+    @JsonProperty
+    private String responsibilities;
+
+    @JsonProperty
+    private String jobSpec;
+
+    // Private constructor to enforce the use of the Builder
+    private JobRole(final Builder builder) {
+        this.id = builder.id;
+        this.roleName = builder.roleName;
+        this.location = builder.location;
+        this.capability = builder.capability;
+        this.band = builder.band;
+        this.closingDate = builder.closingDate;
+        this.status = builder.status;
+        this.description = builder.description;
+        this.responsibilities = builder.responsibilities;
+        this.jobSpec = builder.jobSpec;
+    }
+
+    // Static nested Builder class
+    public static class Builder {
+        private int id;
+        private String roleName;
+        private String location;
+        private String capability;
+        private String band;
+        private Date closingDate;
+        private String status;
+        private String description;
+        private String responsibilities;
+        private String jobSpec;
```

Comment:
Why remove the default constructor and add builder?

Reply:
cause that was a work around to the 7+ parmeters linting check